### DISCUSSION TITLE
Gestion des périmètres

### DISCUSSION THREAD

**Discussion by dbarzin** [dbarzin/deming#41] [Announcements]
Voici ce que je propose de développer pour la gestion des périmètres : 

*  L’administrateur peut gérer les périmètres dans la partie administration de l’application. Cette fonctionnalité est identique à celle de la gestion des domaines. 
* Le nom du périmètre est une chaîne de 32 caractères. Ex. : « Le site principal », « le site secondaire », « l’active directory principal » 
* Un champ périmètre est ajouté à la table contrôle 
* Lors de la planification d’un contrôle, il est possible de spécifier le périmètre 
* La liste des mesures de sécurité ne fait plus référence au dernier contrôles (car il peut y en avoir plusieurs) mais au nombre à réaliser de contrôles. 
* L’affichage d’une mesure de sécurité montre tout ses contrôles associés avec leur périmètres respectifs
* Deux contrôles (non réalisés) portant sur la même mesure de sécurité ne peuvent pas avoir le même périmètre 
* Le périmètre peut être vide, il concerne alors tout le domaine d’application 
* Dans la liste des contrôles, il est possible de filtrer par périmètre 
* La liste des contrôles contient une colonne reprenant le périmètre du contrôle 
* Dans les tableaux de bord, on peut filtrer sur : 
    * le périmètre vide ;
    * tous les périmètres ;
    * un périmètre spécifique.
* Dans les tableaux de bord (point de discussion) : 
    * Les résultats des contrôles sont groupés par mesures, on calcule : 
        * la moyenne pour tous les périmètres (ou ?) 
        * la valeur minimum de tous les périmètres 
(ou ?) 
    * Les résultats des contrôles ne sont pas groupés, on affiche toutes les valeurs. 
  
Cette proposition est ouverte aux commentaires.
@yjacolin @0xlildoudou

**Comment by alexiaauddino** [comment]
Bonjour, 

Merci pour cet outil ! Je me demandais si vous aviez toujours pour projet de développer cette fonctionnalité, qui, il me semble, n'existe pas actuellement dans Deming ? 
Je trouve la fonctionnalité très intéressante, et j'imagine même des accès différents en fonction des périmètres. 

Disponible pour échange à ce sujet !

**Comment by dbarzin** [comment]
Cette fonctionalité est en disponible. Lors de la plannification d'un contrôle, il est possible de spécifier le prérimètre : 

![image](https://github.com/user-attachments/assets/cb679e5d-f021-4163-a72b-c57d8972d81b)

et dans les tableaux de bord, il est possible également de spécifier ce périmètre : 

![image](https://github.com/user-attachments/assets/cc0c3942-51f6-4e08-b76c-25f99741b420)

**Comment by alexiaauddino** [comment]
Merci beaucoup pour votre réponse ! Cela nous aide beaucoup.
- _L’administrateur peut gérer les périmètres dans la partie administration de l’application. Cette fonctionnalité est identique à celle de la gestion des domaines._  - Est ce que c'est possible finalement ? De ce que j'ai vu dans l'outil, on peut créer un périmètre lorsque l'on planifie un contrôle sur une mesure. Est ce possible d'avoir des sous-périmètre associé à un périmètre ? 

Est ce qu'il y a une gestion des accès possible en fonction d'un périmètre donné ? 
Quand on édite un rapport, est ce qu'on peut l'éditer pour un périmètre / sous-périmètre sélectionné ? 

Merci encore pour les réponses, 
Excellente journée !

**Comment by dbarzin** [comment]
Voici quelques précisions qui, je l'espère, vous apporteront davantage de clarté :

- **Sous-périmètres** : Actuellement, l'application ne gère pas directement la notion de "sous-périmètre". Cependant, vous pouvez structurer et organiser les périmètres via un champ texte, ce qui vous permet de les personnaliser en fonction de vos besoins spécifiques. Cela peut être une solution simple et flexible pour simuler des sous-périmètres.

- **Gestion des accès par périmètre** : Il est possible de restreindre l'accès aux contrôles par utilisateur en fonction du périmètre. Les utilisateurs n'auront accès qu'aux contrôles qui leur sont assignés, ce qui vous permet de définir des accès précis en fonction des périmètres définis.

- **Rapports par périmètre** : Actuellement, les rapports ne sont pas générés spécifiquement par périmètre ou sous-périmètre. Cela dit, c'est une fonctionnalité qui peut être implémentée facilement en tant qu'évolution de l'application si cela répond à vos besoins.
